Source/Fuse.Reactive.Expressions/Tests/MathFunctions.Test.uno

```diff
@@ -0,0 +1,231 @@
+			{
+				Assert.AreEqual(1, p.c.Integer);
+				
+				p.b.Float = 3.5f;
+				root.PumpDeferred();
+				Assert.AreEqual(3.0f, p.c.Float);
+				
+				p.a.Float = -9.5f;
+				root.PumpDeferred();
+				Assert.AreEqual(1f, p.c.Float);
+				
+				p.b.Float = -3f;
+				root.PumpDeferred();
+				Assert.AreEqual(-0.5f, p.c.Float);
+			}
+		}
+
+		[Test]
+		public void EvenOdd()
+		{
+			var p = new UX.MathFunctions.EvenOdd();
+			using (var root = TestRootPanel.CreateWithChild(p))
+			{
+				Assert.IsTrue(p.b.Boolean);
+				Assert.IsFalse(p.c.Boolean);
+				
+				p.a.Integer = -3;
+				root.PumpDeferred();
+				Assert.IsFalse(p.b.Boolean);
+				Assert.IsTrue(p.c.Boolean);
+				
+				p.a.Float = 2.6f; //rounds to 3
+				root.PumpDeferred();
+				Assert.IsFalse(p.b.Boolean);
+				Assert.IsTrue(p.c.Boolean);
+				
+				p.a.Float = -2.3f; //rounds to -2
+				root.PumpDeferred();
+				Assert.IsTrue(p.b.Boolean);
+				Assert.IsFalse(p.c.Boolean);
+			}
+		}
+		
+		[Test]
+		public void Alternate()
+		{
+			var p = new UX.MathFunctions.Alternate();
+			using (var root = TestRootPanel.CreateWithChild(p))
+			{
+				Assert.IsFalse(p.b.Boolean);
+				
+				var ts = new[]{ -6, -5, -4, 0, 1, 2, 6 };
+				var fs = new[]{ -7, -3, -2, -1, 3, 4, 5 };
+				for (int i=0; i < ts.Length; ++i)
+				{
+					p.a.Integer = ts[i];
+					root.PumpDeferred();
+					Assert.IsTrue(p.b.Boolean);
+					p.a.Integer = fs[i];
+					root.PumpDeferred();
+					Assert.IsFalse(p.b.Boolean);
+				}
+			}
+		}
+		
+		[Test]
+		public void Simple()
+		{
+			var p = new UX.MathFunctions.Simple();
+			using (var root = TestRootPanel.CreateWithChild(p))
+			{
+				var values = new[]{ 1.0f, 0.5f, 0.0f, -0.55f, 0.7f, -0.4f };
+				for (int i=0; i < values.Length; ++i)
+				{
+					var v = values[i];
+					//first time comes from 1.0 value in the UX on rooting
+					if (i !=0)
+					{
+						p.a.Float = v;
+						root.PumpDeferred();
+					}
+					Assert.AreEqual( Math.Sin(v), p.sin.Float );
+					Assert.AreEqual( Math.Cos(v), p.cos.Float );
+					Assert.AreEqual( Math.Tan(v), p.tan.Float );
+					Assert.AreEqual( Math.Asin(v), p.asin.Float );
+					Assert.AreEqual( Math.Acos(v), p.acos.Float );
+					Assert.AreEqual( Math.Atan(v), p.atan.Float );
+					Assert.AreEqual( Math.Abs(v), p.abs.Float );
+					Assert.AreEqual( Math.Ceil(v), p.ceil.Float );
+					Assert.AreEqual( Math.Floor(v), p.floor.Float );
+					Assert.AreEqual( Math.DegreesToRadians(v), p.degreesToRadians.Float );
+					Assert.AreEqual( Math.RadiansToDegrees(v), p.radiansToDegrees.Float );
+					Assert.AreEqual( Math.Exp(v), p.exp.Float );
+					Assert.AreEqual( Math.Exp2(v), p.exp2.Float );
+					Assert.AreEqual( Math.Fract(v), p.fract.Float );
+					Assert.AreEqual( Math.Sign(v), p.sign.Float );
+					Assert.AreEqual( Math.Round(v), p.round.Float );
+					Assert.AreEqual( Trunc.Op(v), p.trunc.Float );
+					//some have special exceptions where the Inf/NaN value doesn't compare right on some platforms
```

Comment:
I think it was DotNet and also Windows. I didn't think we'd be able to influence NaN handling on any target anyway, plus since the results make no sense in these test conditions, I didn't bother looking further. I don't know which platform is correct, in theory NaN shouldn't compare valid on any of them, but since there's object wrappers involved in some... 

The conditions limit the tests for proper values, skipping NaN tests then.